### PR TITLE
privileges: require process privilege to access metrics_schema

### LINKED ISSUE #27553: Require process privilege to access metrics_schema
## Enhancement

The system database `metrics_schema` contains metrics about the cluster. To visit the load on the cluster, users need to have `PROCESS` privileges.

Besides, another 2 system tables `information_schema.metrics_summary` and `information_schema.metrics_summary_by_label` also contain metrics on the cluster.

### PR DESCRIPTION
**Pull request by djshow832**
### What problem does this PR solve?

Issue Number: close #27553

Problem Summary:
- Require `PROCESS` or `SELECT` privilege to access `metrics_schema`.
- Require `PROCESS` privilege to access `information_schema.metrics_summary` and `information_schema.metrics_summary_by_label`.

### What is changed and how it works?

What's Changed:
If the user doesn't have `process` or `select` privilege, report an error.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
- Require `PROCESS` or `SELECT` privilege to access `metrics_schema`.
- Require `PROCESS` privilege to access `information_schema.metrics_summary` and `information_schema.metrics_summary_by_label`.
```

### CODE REVIEW

File: executor/infoschema_reader.go [hunk 1]
```diff
@@ -128,7 +128,7 @@ func (e *memtableRetriever) retrieve(ctx context.Context, sctx sessionctx.Contex
 		case infoschema.TableKeyColumn:
 			e.setDataFromKeyColumnUsage(sctx, dbs)
 		case infoschema.TableMetricTables:
-			e.setDataForMetricTables(sctx)
+			err = e.setDataForMetricTables(sctx)
```
Comment (crazycs520) [inline executor/infoschema_reader.go:131]: useless change?

File: privilege/privileges/privileges.go [hunk 2]
```diff
@@ -143,16 +143,26 @@ func (p *UserPrivileges) RequestVerification(activeRoles []*auth.RoleIdentity, d
...
 				return true
 			}
 		}
+	case util.MetricSchemaName.L:
+		if infoschema.IsMetricTable(table) {
+			switch priv {
+			case mysql.CreatePriv, mysql.AlterPriv, mysql.DropPriv, mysql.IndexPriv, mysql.InsertPriv, mysql.UpdatePriv, mysql.DeletePriv:
+				return false
+			// PROCESS is the same with SELECT for metrics_schema.
+			case mysql.SelectPriv:
```
[the remark this answers is not in the document]
Comment (djshow832) [reply privilege/privileges/privileges.go:162]: No, it won't lead to privilege escalations. It still requires the `SELECT` privilege on the table, no the database.
But ... I found granting `SELECT` to a specific table will report an error, and so does `performance_schema`.
```sql
mysql> grant select on metrics_schema.up to test_user;
ERROR 1146 (42S02): Table 'metrics_schema.up' doesn't exist
```

**Comment by crazycs520** [comment]
@morgo PTAL

**Review by morgo — APPROVED** [review]

**Pull request merged**